src/synapse/azext_synapse/_params.py

```diff
@@ -0,0 +1,96 @@
+
+    with self.argument_context('synapse spark batch') as c:
+        c.argument('batch_id', options_list=['--id'], arg_group='Spark Batch', help='The id of the spark batch job.')
+        c.argument('job_type', help='The spark batch job type: SCALA, PYTHON or DOTNET. The default value is SCALA.')
+
+    with self.argument_context('synapse spark session') as c:
+        c.argument('session_id', options_list=['--id'], arg_group='Spark Session',
+                   help='The id of the spark session job.')
+        c.argument('num_executors', help='The number of executors.')
+
+    with self.argument_context("synapse spark session-statement") as c:
+        c.argument('statement_id', options_list=['--id'], arg_group="Spark Session-statement",
+                   help='The id of the statement.')
+        c.argument('session_id', help='The id of spark session job.')
+        c.argument('code', help='The code of spark statement.')
+        c.argument('kind', help='The kind of spark statement.')
+
+    # synapse workspace
+    with self.argument_context('synapse workspace') as c:
+        c.argument('resource_group_name', help='The resource group name.')
+        c.argument('workspace_name', arg_type=name_type, help='The workspace name.')
+        c.argument("account_url", help='The data lake storage account url.')
+        c.argument('file_system', help='The file system of the data lake storage account.')
+        c.argument('sql_admin_login_user', help='The sql administrator login user name.')
+        c.argument('sql_admin_login_password', help='The sql administrator login password.')
+        c.argument('tags', arg_type=tags_type, help='The tag of workspace.')
+        c.argument('identity_type', help='The type of managed identity.')
+        c.argument('principal_id', help='The principal id of managed identity.')
+        c.argument('tenant_id', help='The tenant id of managed identity.')
+
+    # synapse spark pool
+    with self.argument_context('synapse spark pool') as c:
+        c.argument('spark_pool_name', arg_type=name_type, help='The spark pool name.')
+        c.argument('workspace_name', help='The workspace name.')
+        c.argument('auto_pause_enabled', help='The flag of enabling auto pause.')
+        c.argument('auto_scale_enabled', help='The flag of enabling auto scale.')
+        c.argument('default_spark_log_folder', help='The default spark log folder.')
+        c.argument('delay_in_minutes', help='The delay time.')
+        c.argument('library_requirements_content', help='The library requirements content.')
+        c.argument('library_requirements_filename', help='The library requirements file name.')
+        c.argument('max_node_count', help='The max node count.')
+        c.argument('min_node_count', help='The min node count.')
+        c.argument('node_count', help='The number of node.')
+        c.argument('node_size_family', help='The node size family.')
+        c.argument('node_size', help='The node size.')

```

Comment:
There are lots of arguments for configurations, consider using arg_group to arrange them if possible so help message will be more organized. See arg_group part in https://github.com/Azure/azure-cli/blob/master/doc/authoring_command_modules/authoring_commands.md

Reply:
fixed

---
In reply to: [387015466](https://github.com/Azure/azure-cli-extensions/pull/1339#discussion_r387015466) [](ancestors = 387015466)